Xamarin.Forms.Core/Picker.cs

```diff
@@ -7,246 +7,249 @@
+        }
+
+        void AddItems(NotifyCollectionChangedEventArgs e)
+        {
+            int index = e.NewStartingIndex < 0 ? Items.Count : e.NewStartingIndex;
+            foreach (object newItem in e.NewItems)
+            {
+                Items.Insert(index++, GetDisplayMember(newItem));
+            }
+        }
+
+        void BindItems()
+        {
+            Items.Clear();
+            foreach (object item in ItemsSource)
+            {
+                Items.Add(GetDisplayMember(item));
+            }
+            UpdateSelectedItem();
+        }
+
+        static object CoerceSelectedIndex(BindableObject bindable, object value)
+        {
+            var picker = (Picker)bindable;
+            return picker.Items == null ? -1 : ((int)value).Clamp(-1, picker.Items.Count - 1);
+        }
+
+        void CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            switch (e.Action)
+            {
+                case NotifyCollectionChangedAction.Move:
+                case NotifyCollectionChangedAction.Replace:
+                case NotifyCollectionChangedAction.Reset:
+                    // Clear Items collection and re-populate it with values from ItemsSource
+                    BindItems();
+                    return;
+                case NotifyCollectionChangedAction.Remove:
+                    RemoveItems(e);
+                    break;
+                case NotifyCollectionChangedAction.Add:
+                    AddItems(e);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        static object GetPropertyValue(object item, string memberPath)
+        {
+            if (item == null)
+            {
+                return null;
+            }
+            // TODO How to handle Nullable types
+            if (IsPrimitive(item) || string.IsNullOrEmpty(memberPath))
+            {
+                return item.ToString();
+            }
+            // Find the property by walking the display member path to find any nested properties
+            string[] propertyPathParts = memberPath.Split('.');
+            object propertyValue = null;
+            foreach (string propertyPathPart in propertyPathParts)
+            {
+                PropertyInfo propInfo = item.GetType().GetTypeInfo().GetDeclaredProperty(propertyPathPart);
+                if (propInfo == null)
+                {
+                    throw new ArgumentException(
+                        $"No property with name '{memberPath}' was found on '{item.GetType().FullName}'");
+                }
+                propertyValue = propInfo.GetValue(item);
+            }
+            return propertyValue;
+        }
+
+        static bool IsPrimitive(object item)
+        {
+            return item is string || item is int || item is double || item is decimal || item is Enum ||
```

Comment:
What about `DateTimeOffset`, `long`, `single`, `float` e.t.c should use item.Getype().GetTypeInfo().IsValueType